### DISCUSSION TITLE
Report - (4th Sem.) Week 5 📢

### DISCUSSION THREAD

**Discussion by JonasEbbinghaus25** [SE-TINF22B6/DHBWhub#638] [Weekly Report]
Greetings and welcome to our thirteenth progress report! We're thrilled to update you on the latest news and developments since last week! 🥳

### Developments on the frontend 🖥

This week, we've made some significant progress on the frontend. This includes updating friend list endpoints and adapting posts to handle 'null' values in images. Additionally, we've added PWA shortcuts for common pages, enhancing user accessibility. Below is the post page and the post page without an image but with a NULL value:

**Functioning registration🥳**

We also pressed ahead with adapting and improving the Sing up dialog. Users are now able to register with DHBWhub. We also use email verification for this, as you can see in the picture of the email we sent in last week's blog. Below are the screenshots of the three stages of the Sing-up dialog:

First step - Enter your email for verification:

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/2bfb93cd-a6e2-4727-8d75-34dacd688a23)

Second step - Enter the verification token that was sent to the new user in the email:

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/c2d3d93c-201e-4897-9b94-56501ae94526)

Third step - Enter the user name and a password:

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/c3c5cad3-7b05-4371-8e94-0919450a315d)

### Developments on the database 💾

Our database has also received some important additions. We've created a log table for likes and a notification table to improve the notification system. These updates will contribute to enhancing the performance and functionality of our application.

### Developments on the backend 🧮

In the backend realm, we've been working on implementing key components, including 'PostDetail' and 'CreateComment'. Furthermore, we've integrated requests for login and signup processes in Postman to streamline interaction with our API.

### Design and Mockups 🎨

This week, we focused on creating mockups for key pages of our project. We designed profiles and user pages to ensure clear navigation and engaging user interactions. Additionally, we crafted mockups for tag overview and imprint pages, prioritizing simplicity and professionalism. These mockups will guide our development process towards a visually appealing and user-friendly application. Below are the new mockups:

User Page:

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/2ae13a5e-e8b5-4a1a-b32d-0f725edf68f2)

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/5c1ff563-8b97-458a-9307-1c69885d567b)

Tag overview page:

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/04c40398-b68d-46d8-b8ea-e67cdb9dfc15)

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/e9f6aa79-0544-4ce6-846b-f76474176bc7)

Imprint page:

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/898ead06-e33e-4a3c-8b85-5b13f4235d86)

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/ca654043-1ff8-41e3-acf4-dfaaa78b1426)

Profile page:

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/64d57032-83cf-429b-ad37-c29856290068)

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/38cef27b-603e-46a0-99b5-f52352eb9eec)

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/75144058/c8a3cb55-e6c3-4711-bd0c-d34c1380b00d)

### Main goals and priorities in the next weeks 📈

For the upcoming sprint, we've set the following goals:

Sprint 8/9: What have we achieved in Sprint 8 ❓ and what is the outlook for Sprint 9❔
✅ Frontend: Updating endpoints and adding PWA shortcuts
✅ Frontend: Function to query whether a user is logged in
✅ Frontend: Enable registration 
✅ Frontend: Fix report window
✅ Design: Create mockup for "Tag overview"-page
✅ Design: Create mockup for privacy policy page
✅ Design: Create mockup for profile page
✅ Design: Create mockup for user page
✅ Design: Create mockup for imprint page
✅ Backend: Verification of the e-mail address
✅ Backend: Enable sending of mails
✅ Backend: Adapt Friendlist
✅ Backend: Implementing 'PostDetail' and 'CreateComment' components
✅ Database: Creating log and notification tables
✅ Database+Backend: Adding login/signup requests in Postman
🚧 Frontend: Adapting the design for login and signup dialogues
🚧 Frontend: Adapt profile page
🚧 Frontend: Create further tests for the essential services
🚧 Frontend: Implement friends page
🚧 Backend: Implement user and post search
🚧 Backend: Find posts from a certin user 
🚧 Backend: Refine Postrequests
🚧 Backend: GoogleAuth
🚧 Backend and Database: Find popular Tags
🚧 Write Imprint
🚧 Write FAQ
🚧 Remove Herobrine

Until next week! 💯

Your DHBWhub Team 🚀

**Comment by BenjaminB109** [comment]
Dear DHBWhub Team,
I am very glad to see that you have successfully implemented the registration with email verification feature. The new mockups for key pages such as profiles and tag overview look really good.

**Comment by Maximon132** [comment]
Hey everyone,
I really like the idea of having an email verification feature. I also appreciate the progress you've made on the frontend and how minimalistic it looks.

**Comment by HowDoIUseGit** [comment]
Dear DHBWhub,

Your exmail verification looks awesome. I also like the idea of the PWA shortcuts
